src/core/xfa/factory.js

```diff
@@ -22,28 +22,61 @@ class XFAFactory {
     try {
       this.root = new XFAParser().parse(XFAFactory._createDocument(data));
       this.form = new Binder(this.root).bind();
-      this._createPages();
     } catch (e) {
       console.log(e);
     }
   }
 
+  isValid() {
+    return this.root && this.form;
+  }
+
   _createPages() {
-    this.pages = this.form[$toHTML]();
-    this.dims = this.pages.children.map(c => {
-      const { width, height } = c.attributes.style;
-      return [0, 0, parseInt(width), parseInt(height)];
-    });
+    try {
+      this.pages = this.form[$toHTML]();
+      this.dims = this.pages.children.map(c => {
+        const { width, height } = c.attributes.style;
+        return [0, 0, parseInt(width), parseInt(height)];
+      });
+    } catch (e) {
+      console.log(e);
```

Comment:
Please use `warn` instead, since we should never use `console.log(...)` *directly* in code residing in the `src/` folder.